tests/unit/customizations/s3/test_utils.py

```diff
@@ -486,6 +486,16 @@ def test_out_file_with_encoding_attribute(self):
         uni_print(u'\u2713', out)
         self.assertEqual(buf.getvalue(), u'\u2713'.encode('utf-8'))
 
+    def test_encoding_with_encoding_none(self):
+        '''When the output of the aws command is being piped,
+        the ``encoding`` attribute of ``sys.stdout`` is ``None``.'''
+        buf = io.BytesIO()
+        out = io.TextIOWrapper(buf, encoding=None)
+        uni_print(u'SomeChars\u2713\u2714OtherChars', out)
+        # Unicode characters get replaced with their
+        # UTF-8 byte value.
+        self.assertEqual(buf.getvalue(), b'SomeChars\xe2\x9c\x93\xe2\x9c\x94OtherChars')
```

Comment:
Okay. I found out (I think). `TextIOWrapper` sets `encoding` to `UTF-8` when is passed `encoding=None`, so this test never hits the `UnicodeEncodeError` branch of `uni_print.

I'm gonna update this branch to fix it.